iris-mpc-cpu/src/execution/hawk_main.rs

```diff
@@ -561,21 +573,27 @@ impl HawkResult {
     }
 
     fn merged_results(&self) -> Vec<u32> {
+        let match_ids = self.match_ids();
         self.connect_plans.0[0]
             .iter()
-            .map(|plan| {
-                plan.as_ref()
-                    .map_or(0, |plan| plan.inserted_vector.to_serial_id())
+            .enumerate()
+            .map(|(idx, plan)| match plan {
+                Some(plan) => plan.inserted_vector.to_serial_id(),
+                None => match_ids[idx][0],
```

Comment:
This assumes that there is at least one match, based on the absence of an insertion. This could be more explicit or more defensive with `.first().unwrap_or(0)` as this may not always work (maybe because of "persistence" settings and what not).

Reply:
wouldn't then be a plan if there is no match? that was my assumption, maybe wrong.

Reply:
returning just 0 is not safe, we should always either have a match and return the corresponding index or return the newly inserted entry idx.

Reply:
There is a config `disable persistence` and also requests with `skip persistence`. Not sure how it is all supposed to work with `merged_results` though. But probably the field `is_matches` will be more reliable than `connect_plans`.

https://github.com/worldcoin/iris-mpc/blob/2a1162b7c8b622bf472a4b4069151a27b1bc82d1/iris-mpc-cpu/src/execution/hawk_main.rs#L766

https://github.com/worldcoin/iris-mpc/blob/2a1162b7c8b622bf472a4b4069151a27b1bc82d1/iris-mpc-common/src/job.rs#L59

Reply:
yeah definitely point taken, that we should make this more robust.